celery.go

```diff
@@ -247,13 +254,37 @@ func (a *App) executeTask(ctx context.Context, m *protocol.Task) (err error) {
 		}
 	}()
 
-	task := a.task[m.Name]
+	task, ok := a.task.Load(m.Name)
+	if !ok {
+		return fmt.Errorf("unregistered task: %s", m.Name)
+	}
+	taskF, ok := task.(TaskF)
+	if !ok {
+		return fmt.Errorf("invalid task: %s", m.Name)
+	}
 	// Use middlewares if a client provided them.
 	if a.conf.chain != nil {
-		task = a.conf.chain(task)
+		taskF = a.conf.chain(taskF)
 	}
 
 	ctx = context.WithValue(ctx, ContextKeyTaskName, m.Name)
 	p := NewTaskParam(m.Args, m.Kwargs)
-	return task(ctx, p)
+	return taskF(ctx, p)
+}
+
+// update broker with new queues
+func (a *App) observe() {
```

Comment:
Ok `taskQueue` abd `queueCount` are thread-safe by themselves, but I wonder whether there needs to be some lock so all of this happens atomically?

Reply:
I dont think so, if you find example where this might break let me know.